cancom/services/dynamic-cloud/resource_vpc_project.go

```diff
@@ -0,0 +1,264 @@
+
+	err = d.Set("limits", resp.Spec.Limits)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.Set("name", resp.Metadata.Name)
+	d.Set("openstack_uuid", resp.Spec.OpenstackUuid)
+	d.Set("project_comment", resp.Spec.ProjectComment)
+	d.Set("tenant", resp.Metadata.Tenant)
+
+	humanUsers, err := getHumanUsers(resp.Metadata.Tenant, resp.Metadata.Name, resp.Spec.ProjectUsers)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	err = d.Set("users", usersToSet(humanUsers))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+func resourceVpcProjectUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c, err := meta.(*client.CcpClient).GetService("dynamic-cloud")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tflog.Info(ctx, "Updating VPC Project")
+	vpcProjectShortid := d.Id()
+	users := setToUsers(d.Get("users"))
+
+	// get VPC Project to get any serviceUsers created in the project
+	resp, err := (*client_dynamiccloud.Client)(c).GetVpcProject(vpcProjectShortid)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if resp == nil {
+		d.SetId("")
+		return diag.Errorf("error updating VPC Project. VPC Project NotFound")
+	}
+	serviceUsers, err := getServiceUsers(resp.Metadata.Tenant, resp.Metadata.Name, resp.Spec.ProjectUsers)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tflog.Info(ctx, "Concat svc users", map[string]interface{}{"usersForBody": append(users, serviceUsers...)})
+	_, err = (*client_dynamiccloud.Client)(c).UpdateVpcProjectUsers(vpcProjectShortid, append(users, serviceUsers...))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate)-time.Minute, func() *resource.RetryError {
+		resp, err := (*client_dynamiccloud.Client)(c).GetVpcProject(vpcProjectShortid)
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("error describing VPC Project: %s", err))
+		}
+		if resp == nil {
+			return resource.NonRetryableError(fmt.Errorf("error describing VPC Project. VPC Project 'NotFound'"))
+		}
+		if resp.Status.Phase != "Ready" {
+			tflog.Info(ctx, "Waiting for VPC Project to finish updating")
+			return resource.RetryableError(fmt.Errorf("VPC Project is still transitioning with phase '%s'", resp.Status.Phase))
+		}
+		return nil
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceVpcProjectRead(ctx, d, meta)
+}
+
+func resourceVpcProjectDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c, err := meta.(*client.CcpClient).GetService("dynamic-cloud")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tflog.Info(ctx, "Deleting VPC Project")
+	var diags diag.Diagnostics
+	vpcProjectShortid := d.Id()
+
+	err = (*client_dynamiccloud.Client)(c).DeleteVpcProject(vpcProjectShortid)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutDelete)-time.Minute, func() *resource.RetryError {
```

Comment:
Tested, got an error.
cancom_dynamic_cloud_vpc_project.name: Still destroying... [id=9SHbs566yAS6fhjqfhZpoU, 14m1s elapsed]
╷
│ Error: VPC Project is still transitioning with phase 'Deleting'
│ 


Maybe adjust the timeout.  After 16 min, the backend reports stillt the status 
phase": "Deleting",

Reply:
Thanks you or testing the changes in this PR.
This was not a bug in the provider. We changed how the backend handles deletion and forget to change a reference on the callback telling us the deletion finished.
It's fixed now.